### PR TITLE
[Issue - 506] CarbonSqlParser is not validating the duplicate column …

### PR DESCRIPTION
**Pull request by mohammadshahidkhan**
…name if they are in different case in create table command

Modification:
Now in case duplicate columns  system will throw error while creating table.

### CODE REVIEW

File: integration/spark/src/main/scala/org/apache/spark/sql/CarbonSqlParser.scala [hunk 1]
```diff
@@ -362,7 +362,10 @@ class CarbonSqlParser()
         case list@Token("TOK_TABCOLLIST", _) =>
           val cols = BaseSemanticAnalyzer.getColumns(list, true)
           if (cols != null) {
-
+            var dup = cols.asScala.groupBy(x => x) filter { case (_, colList) => colList.size > 1 }
+            if (dup.size > 0) {
```
Comment (nareshpr) [inline integration/spark/src/main/scala/org/apache/spark/sql/CarbonSqlParser.scala]: column name is same, datatype is different, will this fix work ?

**Comment by mohammadshahidkhan** [comment]
Fixed review comments

File: integration/spark/src/main/scala/org/apache/spark/sql/CarbonSqlParser.scala (hunk 1 above)
Comment (mohammadshahidkhan) [inline integration/spark/src/main/scala/org/apache/spark/sql/CarbonSqlParser.scala]: handled now comparing on the column name

**Comment by ravipesala** [comment]
@mohammadshahidkhan  Please rebase the code

File: integration/spark/src/main/scala/org/apache/spark/sql/CarbonSqlParser.scala [hunk 2]
```diff
@@ -362,7 +362,18 @@ class CarbonSqlParser()
         case list@Token("TOK_TABCOLLIST", _) =>
           val cols = BaseSemanticAnalyzer.getColumns(list, true)
           if (cols != null) {
-
+            val dupColsGrp = cols.asScala
+              .groupBy(x => x.getName) filter { case (_, colList) => colList
```
Comment (ravipesala) [inline integration/spark/src/main/scala/org/apache/spark/sql/CarbonSqlParser.scala]: groupBy is not case sensitive right? How the duplicate columns with different case letters are grouped together?

Comment (mohammadshahidkhan) [inline integration/spark/src/main/scala/org/apache/spark/sql/CarbonSqlParser.scala]: val cols = BaseSemanticAnalyzer.getColumns(list, true) is returning the cols in lower case only so no need to handle the case.

**Comment by mohammadshahidkhan** [comment]
rebase done

**Comment by ravipesala** [comment]
LGTM

**Pull request merged**